src/indexer/utils/evm/index.ts

```diff
@@ -91,35 +101,40 @@ export async function getDecodedLogs(
     }
 
     case EventType.FEE_COLLECTED: {
-      const feeCollected = await parseFeeCollected(decodedLog, provider, domain.nativeTokenSymbol, log)
+      const feeCollected = await parseFeeCollected(decodedLog, provider, fromDomain.nativeTokenSymbol, log)
       decodedLogs.feeCollected.push(feeCollected)
       break
     }
   }
 }
 
-export function parseDeposit(
-  domain: Domain,
+export async function parseDeposit(
+  fromDomain: Domain,
+  toDomain: Domain,
   log: Log,
   decodedLog: LogDescription,
   txReceipt: TransactionReceipt,
   blockUnixTimestamp: number,
   resourceMap: Map<string, EvmResource>,
-): DecodedDepositLog {
+): Promise<DecodedDepositLog> {
   const resourceType = resourceMap.get(decodedLog.args.resourceID as string)?.type || ""
   const resourceDecimals = resourceMap.get(decodedLog.args.resourceID as string)?.decimals || 18
-
   const arrayifyData = getBytes(decodedLog.args.data as BytesLike)
-  const filtered = arrayifyData.filter((_, idx) => idx + 1 > 65)
-  const hexAddress = hexlify(filtered)
+
+  let destination = ""
```

Comment:
Move to a separate function.
Would be nice if slowly tart testing stuff so also test that function.

Reply:
(if possible parseDeposit as well)